src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -88,7 +96,14 @@ public void mergeOneField(final FieldInfo fieldInfo, final MergeState mergeState
         // This will ensure that we are merging the FlatIndex during force merge.
         flatVectorsWriter.mergeOneField(fieldInfo, mergeState);
         // For merge, pick values from flat vector and reindex again. This will use the flush operation to create graphs
-        trainAndIndex(fieldInfo, this::getKNNVectorValuesForMerge, NativeIndexWriter::mergeIndex, mergeState);
+        trainAndIndex(
```

Comment:
Can we add a unit test?

Reply:
The class seems to only have a component test. Will try to cover it in the existing component test and create an issue to add unit test around this so it can be taken as a follow up

Reply:
@heemin32 do we need UTs to check if the log is coming?

Reply:
Not about logging but about adding build time in metrics.

Reply:
> The class seems to only have a component test. Will try to cover it in the existing component test and create an issue to add unit test around this so it can be taken as a follow up

I believe, all these static method is making it harder to write a unit test. I wish we can rely more on dependency injection using Guice.

Reply:
> I believe, all these static method is making it harder to write a unit test. I wish we can rely more on dependency injection using Guice.

The stats needs to be static as of now as the are gathered at node level. and if we want to make these things work via DI then its a change which is out of scope of this PR.

Reply:
I know. Just sharing my long term desire and testing the water :)